packages/attack-http-requests/src/index.ts

```diff
@@ -0,0 +1,46 @@
+import Mitm from 'mitm';
+
+export interface HTTPAttackOptions {
+  ignoreURLpatterns?: string[];
+}
+
+export default class HTTPAttack {
+  static configure(opts: HTTPAttackOptions): () => HTTPAttack {
+    return () => {
+      const attack = new HTTPAttack(opts);
+      return attack;
+    };
+  }
+
+  ignoreURLpatterns: string[];
+  stopped = false;
+
+  constructor({ ignoreURLpatterns = [] }: HTTPAttackOptions = {}) {
+    this.ignoreURLpatterns = ignoreURLpatterns;
+  }
+
+  start(): void {
+    const mitm = Mitm();
+
+    mitm.on('connect', (socket, opts) => {
+      // https://github.com/moll/node-mitm/issues/16
+      // eslint-disable-next-line
+      const requestedURL: string = opts?.url?.href;
+
+      const shouldIgnore = this.ignoreURLpatterns
+        .map((needle: string) => requestedURL.includes(needle))
+        .includes(true);
+
+      if (this.stopped || shouldIgnore) socket.bypass();
+    });
+
+    mitm.on('request', (_, res) => {
+      res.statusCode = 500;
```

Comment:
we could allow setting the status in the attack configuration, also consider other errors (e.g. connection errors) and latency/timeouts specific to the connection

Reply:
Happy to start with just an immediate 5xx error for now, but agreed for long term future!